persistence/jpa/src/main/java/org/infinispan/persistence/jpa/JpaStore.java

```diff
@@ -321,6 +312,56 @@ public boolean delete(Object key) {
    }
 
    @Override
+   public void deleteBatch(List<Object> keys) {
+      EntityManager em = emf.createEntityManager();
+      try {
+         EntityTransaction txn = em.getTransaction();
+         long txnBegin = timeService.time();
+         try {
+            txn.begin();
+            for (Object key : keys) {
```

Comment:
I think we should be able to use a `CriteriaDelete` instead since we don't require cascading deletes. This should do the batching for us automatically, instead of having to find and remove each entity.